### ISSUE TITLE
`wpt-chrome-dev-results{,-without-changes}` unable to install chromedriver

### ISSUE THREAD

**Issue by jonathan-j-lee** [web-platform-tests/wpt#42545]
This seems to afflict all [recent](https://community-tc.services.mozilla.com/tasks/b4FisPZzTW6q3uNAfqVFPQ) [pull](https://community-tc.services.mozilla.com/tasks/PURnLqwSRACtMevY9AMlpw) [requests](https://community-tc.services.mozilla.com/tasks/JECx9F6WT9WoIirwn1qV-A):

```
  ERROR: Command errored out with exit status 1:
   command: /home/test/web-platform-tests/_venv3/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-0ejqc2c9/mozlog/setup.py'"'"'; __file__='"'"'/tmp/pip-install-0ejqc2c9/mozlog/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-y1g6o5k3
       cwd: /tmp/pip-install-0ejqc2c9/mozlog/
  Complete output (6 lines):
  usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: setup.py --help [cmd1 cmd2 ...]
     or: setup.py --help-commands
     or: setup.py cmd --help
  
  error: invalid command 'bdist_wheel'
  ----------------------------------------
  ERROR: Failed building wheel for mozlog
 0:00.00 INFO Set tests_root to /home/test/web-platform-tests
 0:00.00 INFO Set metadata_root to /home/test/web-platform-tests
 0:00.00 INFO Set manifest_update to True
 0:00.00 INFO Set manifest_download to True
 0:08.22 INFO Identified 2 affected tests
 0:08.38 INFO Downloading Mojo bindings from https://storage.googleapis.com/chrome-wpt-mojom/120.0.6062.2/linux64/mojojs.zip
 0:08.45 INFO MojoJS enabled automatically (mojojs_path: /home/test/web-platform-tests/_venv3/mojojs/gen)
 0:08.48 INFO Searching for ChromeDriver downloads for version 120.0.6062.2.
 0:08.56 INFO No builds found for version 120.0.6062.
Traceback (most recent call last):
  File "./wpt", line 10, in <module>
    wpt.main()
  File "/home/test/web-platform-tests/tools/wpt/wpt.py", line 233, in main
    rv = script(*args, **kwargs)
  File "/home/test/web-platform-tests/tools/wpt/run.py", line 1012, in run
    setup_cls, wptrunner_kwargs = setup_wptrunner(venv, **kwargs)
  File "/home/test/web-platform-tests/tools/wpt/run.py", line 984, in setup_wptrunner
    setup_cls.setup(kwargs)
  File "/home/test/web-platform-tests/tools/wpt/run.py", line 249, in setup
    self.setup_kwargs(kwargs)
  File "/home/test/web-platform-tests/tools/wpt/run.py", line 463, in setup_kwargs
    webdriver_binary = self.browser.install_webdriver(
  File "/home/test/web-platform-tests/tools/wpt/browser.py", line 1255, in install_webdriver
    chromedriver_path = self.install_webdriver_by_version(version, dest, channel)
  File "/home/test/web-platform-tests/tools/wpt/browser.py", line 1267, in install_webdriver_by_version
    raise DownloadNotFoundError(
tools.wpt.browser.DownloadNotFoundError: No ChromeDriver download found to match browser version 120.0.6062.2 dev
[taskcluster 2023-10-13 22:51:25.394Z] === Task Finished ===
[taskcluster 2023-10-13 22:51:26.142Z] Unsuccessful task run with exit code: 1 completed in 104.534 seconds
```

Speculative CC: @past @jcscottiii @DanielRyanSmith @mathiasbynens

**Comment by jonathan-j-lee** [comment]
Looks like the version of `google-chrome-unstable` installed (`120.0.6062*`) is not listed in https://googlechromelabs.github.io/chrome-for-testing/latest-patch-versions-per-build-with-downloads.json, which currently has `120.0.6061` and `120.0.6065`.

**Comment by mathiasbynens** [comment]
> Looks like the version of `google-chrome-unstable` installed (`120.0.6062*`) is not listed in https://googlechromelabs.github.io/chrome-for-testing/latest-patch-versions-per-build-with-downloads.json, which currently has `120.0.6061` and `120.0.6065`.

It seems to be there now. [The text-based endpoint](https://github.com/GoogleChromeLabs/chrome-for-testing#other-api-endpoints) at [`https://googlechromelabs.github.io/chrome-for-testing/LATEST_RELEASE_120.0.6062`](https://googlechromelabs.github.io/chrome-for-testing/LATEST_RELEASE_120.0.6062) also seems to work as expected.

I suspect this was a timing issue, where there was some delay between `google-chrome-unstable` (which is not Chrome for Testing!) getting updated and the CfT data files getting updated.

There are two possible solutions:

1. Avoid `google-chrome-unstable` and instead use Chrome for Testing’s `chrome` binaries corresponding to the Canary channel. That way you’ll always have `chrome` + `chromedriver` binaries of the same version.
2. If you must use `google-chrome-unstable` or any other non-CfT Chrome binaries, then amend your version selection logic to fall back to the `latest-versions-per-milestone` endpoint as described on <https://chromedriver.chromium.org/downloads/version-selection>.

**Comment by jonathan-j-lee** [comment]
This is [happening again](https://community-tc.services.mozilla.com/tasks/csaJct4SQyia0TdTek2CBw/runs/0/logs/live/public/logs/live.log):

```
 0:08.86 INFO Searching for ChromeDriver downloads for version 120.0.6090.0.
 0:08.89 INFO No builds found for version 120.0.6090.
Traceback (most recent call last):
  File "./wpt", line 10, in <module>
    wpt.main()
  File "/home/test/web-platform-tests/tools/wpt/wpt.py", line 233, in main
    rv = script(*args, **kwargs)
  File "/home/test/web-platform-tests/tools/wpt/run.py", line 1012, in run
    setup_cls, wptrunner_kwargs = setup_wptrunner(venv, **kwargs)
  File "/home/test/web-platform-tests/tools/wpt/run.py", line 984, in setup_wptrunner
    setup_cls.setup(kwargs)
  File "/home/test/web-platform-tests/tools/wpt/run.py", line 249, in setup
    self.setup_kwargs(kwargs)
  File "/home/test/web-platform-tests/tools/wpt/run.py", line 463, in setup_kwargs
    webdriver_binary = self.browser.install_webdriver(
  File "/home/test/web-platform-tests/tools/wpt/browser.py", line 1255, in install_webdriver
    chromedriver_path = self.install_webdriver_by_version(version, dest, channel)
  File "/home/test/web-platform-tests/tools/wpt/browser.py", line 1267, in install_webdriver_by_version
    raise DownloadNotFoundError(
tools.wpt.browser.DownloadNotFoundError: No ChromeDriver download found to match browser version 120.0.6090.0 dev
```

https://github.com/GoogleChromeLabs/chrome-for-testing/blob/111ed8b3d35d402021bbf3edf45ef3cc324de4f5/data/latest-patch-versions-per-build-with-downloads.json#L12312-L12362

**Comment by past** [comment]
Fixed by #42853.
